src/oop/object_interactions/medium/q6/MainTest.java

```diff
@@ -0,0 +1,210 @@
+            _assertEquals(seat, testClass.getFieldValue(classInstance, maxSeatsAttributeName),
+                    "You have modified the provided constructor on the " + className + " class. Please revert it to the original state.");
+            _assertEquals(0, testClass.getFieldValue(classInstance, numPassengersAttributeName),
+                    "You have modified the provided constructor on the " + className + " class. Please revert it to the original state.");
+            Object[][] arguments = {
+                    {seat, int.class},
+                    {pass, int.class}
+            };
+            classInstance = testClass.createInstance(arguments);
+            _assertEquals(seat, testClass.getFieldValue(classInstance, maxSeatsAttributeName),
+                    "You have modified the provided constructor on the " + className + " class. Please revert it to the original state.");
+            _assertEquals(pass, testClass.getFieldValue(classInstance, numPassengersAttributeName),
+                    "You have modified the provided constructor on the " + className + " class. Please revert it to the original state.");
+            int updateSeat = (int) test[num][2];
+            int updatePass = (int) test[num][3];
+            Object output = testClass.callMethod(getMaxSeatsMethodName, classInstance);
+            _assertEquals(seat, output,
+                    "You have modified the provided " + getMaxSeatsMethodName + " method on the " + className + " class. Please revert it to the original state.");
+            output = testClass.callMethod(getNumPassengersMethodName, classInstance);
+            _assertEquals(pass, output,
+                    "You have modified the provided " + getNumPassengersMethodName + " method on the " + className + " class. Please revert it to the original state.");
+            testClass.callMethod(setMaxSeatsMethodName, new Object[][]{{updateSeat, int.class}}, classInstance);
+            _assertEquals(updateSeat, testClass.getFieldValue(classInstance, maxSeatsAttributeName),
+                    "You have modified the provided " + setMaxSeatsMethodName + " method on the " + className + " class. Please revert it to the original state.");
+            testClass.callMethod(setNumPassengersMethodName, new Object[][]{{updatePass, int.class}}, classInstance);
+            _assertEquals(updatePass, testClass.getFieldValue(classInstance, numPassengersAttributeName),
+                    "You have modified the provided " + setNumPassengersMethodName + " method on the " + className + " class. Please revert it to the original state.");
+        }
+    }
+
+    private static Stream<Arguments> inputProvider() {
+        return Stream.of(
+                Arguments.of(20, 10, 40, 35),
+                Arguments.of(10, 20, 15, 30),
+                Arguments.of(150, 200, 100, 50),
+                Arguments.of(20, 20, 10, 20)
```

Comment:
Some of these should use not as nice numbers.

Reply:
Also, for the last test case, I would not have it so there are fewer seats than passengers on the plane. This could potentially cause the case to fail depending on the student's code and not test what you want to.

Reply:
And we should also have a case where there are not enough passengers on the busy plane to fill the other plane.